rfcs/45-stageless.md

```diff
@@ -0,0 +1,725 @@
+- states (simplified, no longer purely run-criteria powered)
+- fixed time steps (no longer a run criterion)
+- exclusive systems (no longer special-cased)
+- command processing (now performed in a `flush_commands` exclusive system)
+- system labels and system sets (merged, can now be directly configured)
+- stages (yoten)
+
+### Scheduling overview
+
+Systems in Bevy are stored in a `Schedule`: a collection of **configured systems**, each of which may belong to any number of **system sets**.
+In each iteration of the schedule (typically corresponding to a single frame), the `App`'s `runner` function will run the schedule,
+causing the **scheduler** to run the systems in parallel using a strategy that respects all of the configured constraints.
+If these constraints cannot be met (for example, a system may want to run both before and after another system), the schedule is said to be **unsatisfiable**, and the scheduler will panic.
+Schedules can be nested and branched from within **exclusive systems** (which have mutable access to the entire `World`), allowing you to encode arbitrarily complex control flow using ordinary Rust constructs.
+
+By default, each new `Schedule` is entirely unordered: systems will be selected in an arbitrary order (which is not guaranteed to be stable across passes of the schedule) and run if and only  if all of the data that they must access are free.
+Just like with standard borrow checking, multiple systems can read from the same data at once, but writing to the data requires an exclusive lock.
+Systems which cannot be run in parallel are said to be **incompatible**.
+
+### The `SystemRegistry` resource stores multiple schedules
+
+Each `World` can store multiple `Schedules` with a `SystemRegistry` resource: each with their own `impl ScheduleLabel` type, allowing you to cleanly execute coherent blocks of logic when the need arises.
+For example, by default, each app stores both a startup and main schedule: the former runs only once on app creation, while the latter loops endlessly.
+
+The main and startup schedules can be accessed using the `CoreSchedule::Main` and `CoreSchedule::Startup` labels respectively.
+However, schedules are a surprisingly powerful and flexible tool: they can be used to handle initialization and cleanup logic when working with states or used ad-hoc to run game logic in complex patterns.
+By default, systems are added to the main schedule.
+You can control this by adding the `.in_set_schedule(ScheduleLabel::Variant)` system descriptor to your system or system set.
+
+#### Startup systems
+
+Startup systems are stored in their own schedule, with the `CoreSchedule::Startup` label.
+When using the runner added by `MinimalPlugins` and `DefaultPlugins`, this schedule will run exactly once on app startup.
+
+You can add startup systems with the `.add_startup_system(on_startup)` method on `App`, which is simply sugar for `.add_system(on_startup.in_set_schedule(CoreSchedule::Startup))`.

```

Comment:
Yep, I agree. I'll correct this. I didn't add any shortcut methods in my implementation PR.

We may want to have some like this to make migration easier, idk.

Reply:
Yeah, we should add that to open questions IMO. The unity is good, but startup systems are pretty common (although arguably a lot less so in serious apps, since states tend to work better).